docs/docs/topics/groups.mdx

```diff
@@ -45,34 +45,125 @@ query {
+This design solves the problem of maintaining consistency between generated objects and their logical groupings. When a generator creates 100 interfaces based on a template, they automatically belong to a group that tracks their origin and purpose.
 
-## Group types
+### Query groups
+
+Query groups (`CoreGraphQLQueryGroup`) introduce the concept of dynamic membership based on query results. The group's membership changes as the underlying data changes, maintaining a live view of objects matching specific criteria.
+
+This approach reflects a shift from imperative (explicitly adding members) to declarative (defining membership criteria) group management.
+
+## The visibility model
+
+Infrahub distinguishes between two visibility levels for groups:
 
 ### Default groups
 
-These are user-managed groups, meaning they are created and managed by the user.
-Use Cases: Ideal for organizing and managing resources, users, or devices according to the user's specific needs.
+User-managed groups appear in the standard interface and are intended for direct interaction. These groups represent the organizational structure that users actively maintain and reference in their daily operations.
 
 ### Internal groups
 
-These are system-managed groups, automatically generated by Infrahub to support internal processes such as tracking nodes for artifact definitions.
-In the groups manager view in the UI, a toggle is available to show or hide internal groups. By default, internal groups are hidden to reduce visual clutter and prevent confusion, as they are typically used behind the scenes by Infrahub.
+System-managed groups operate behind the scenes, supporting Infrahub's internal processes. They track relationships that the system needs but users don't typically interact with directly.
+
+The decision to hide internal groups by default reflects a focus on user experience - reducing visual complexity while maintaining the power of the underlying system. Users can toggle visibility when needed for debugging or advanced operations.
+
+## Groups as integration points
+
+Groups serve as a bridge between different Infrahub features, creating a unified way to target operations across the system.
+
+### Artifact generation
+
+When an artifact definition needs to generate configurations for multiple devices, it targets a group. This design decouples the artifact logic from specific device lists, making definitions more reusable and maintainable.
+
+### Transformation targets
+
+Transformations can operate on entire groups, applying consistent changes across all members. The group becomes the unit of operation, simplifying bulk updates and ensuring consistency.
+
+### Check definitions
+
+Validation checks can target groups, running the same validation logic across all members. This approach scales validation from individual objects to entire collections without duplicating logic.
+
+## Design philosophy
```

Comment:
Not sure about this section and the next one (`## Performance considerations`), I think we should remove them